R/checks.R

```diff
@@ -161,10 +168,19 @@ CheckPredictionVariables <- function(object, newdata)
 
                 if (length(level.counts) != 0) # some newdata instances have new levels
                 {
+                    # Use label if available
+                    factor.label <- Labels(object$model, names.to.lookup = train.factor)
+                    if (outliers.removed) # Check if all occurences of new.levels were removed as outliers
+                    {
+                        in.training <- new.levels %in% training[[train.factor]]
+                        in.estimation <- new.levels %in% object$estimation.data[[train.factor]]
+                        warning.msg <- checkPredictionWarningMessage(factor.label, names(level.counts), level.counts,
+                                                                     outliers.removed = in.estimation && !in.training)
+                    } else
+                        warning.msg <- checkPredictionWarningMessage(factor.label, names(level.counts), level.counts)
```

Comment:
Does this function need to be four arguments if one of them is always just the names attribute of another arguments?

Reply:
Good point, I'll reduce it to three.